WorldModel/WorldModel/Core/CoreFunctions.aslx

```diff
@@ -634,5 +632,70 @@
     }
     return (d)
   </function>
+  
+  <function name="InitiateTranscript" parameters="s">
+    filename = GetWindowsSafeName(s)
+    game.transcriptname = GetWindowsSafeName(game.gamename)
+    if (not filename = "") {
+      JS.eval ("transcriptName = '" + filename + "';")
+      //JS.setTranscriptName(filename)
+      game.transcriptname = filename
+    }
+    msg ("> " + game.transcriptname)
+    EnableTranscript
+  </function>
+
+  <function name="EnableTranscript"><![CDATA[
+    if (GetBoolean (game, "notranscript")){
+      KillTranscript
+    }
+    else {
+      if (not HasAttribute (game, "transcriptname")){
+        game.transcriptname = GetWindowsSafeName(game.gamename)
+      }
+      game.savetranscript = true
+      JS.enableTranscript(game.transcriptname)
+      msg ("<br/>" + DynamicTemplate("TranscriptEnabledMessage"))
+      JS.printTranscriptDateAndTime()
+    }
+  ]]></function>
+  
+  <function name="DisableTranscript">
+    game.savetranscript = false
+    JS.disableTranscript()
+  </function>
+  
+  <function name="KillTranscript">
+    game.notranscript = true
+    JS.killTranscript()
+    DisableTranscript
+  </function>
+  
+
+  <function name="UpdateTranscriptString" parameters="data">
+    if (not HasAttribute(game,"transcriptstring")){
+      game.transcriptstring = ""
+    }
+    game.transcriptstring = game.transcriptstring + data
+  </function>
+  
+  <function name="IsDesktop" type="boolean">
+    return (StartsWith(GetFileURL(""), "quest"))
+  </function>
+  
+  <function name="GetWindowsSafeName" parameters="dirty" type="string"><![CDATA[
+    if (StartsWith(dirty, ".")) {
+      dirty = Mid (dirty, 2, LengthOf (dirty) -1)
+    }
+    clean = Trim(Replace(Replace(Replace(Replace(Replace(Replace(Replace(Replace(Replace(dirty, "\"", "''"), "<", "_"), ">", "_"), ":", "_"), "/", "_"), "\\", "_"), "|", "_"), "?", "_"), "*", "_"))
+    changed_something = false
+    if (not clean = dirty) {
+      changed_something = true
+    }
+    if (StartsWith(clean, ".")) {
+      clean = GetWindowsSafeName(clean)
+    }
+    return (clean)
+  ]]></function>

```

Comment:
This feels like the wrong place to be cleaning up a Windows filename? I'd have thought that should be in the desktop player VB code.

Reply:
I had it that way for a short while, but the WebPlayer allows downloading the transcript, too; so, I thought of this to handle it all in one place.

I don't really like it here, either, but I couldn't think of a way to do it without having separate scripts for each player platform. I wouldn't mind coding it that way, if that's the best way to handle things.

Reply:
Dumb question: Will the browser automatically handle safe filenames on its own?

I can't find anything saying so, but this test in Firefox created a file with a Windows-safe name:

```js
function saveFile(fileName,urlFile){
    var a = document.createElement("a");
    a.style = "display: none";
    document.body.appendChild(a);
    a.href = urlFile;
    a.download = fileName;
    a.click();
    window.URL.revokeObjectURL(url);
    a.remove();
}

var stringToSave = `Lorem ipsum
foo bar too`;
var saveFileData = new Blob([stringToSave], {type: "text/plain"});
var saveUrl = window.URL.createObjectURL(blobData);
saveFile('this: <is not a safe \file*name?.txt')
```

![image](https://github.com/user-attachments/assets/7d54085c-9ba8-4c5c-b4cf-072f0d462482)

Reply:
Testing proved the WebPlayer does not need a script to "clean" the filename. The browser handles that when downloading.

All the code concerning this has been moved to **PlayerHTML.vb**, and this bit is ready for re-review. (In fact, this PR is now ready for another review.)